src/main/java/com/google/jenkins/plugins/k8sengine/KubeConfig.java

```diff
@@ -35,38 +36,41 @@
   private static final String API_VERSION = "v1";
   private static final String CONFIG_KIND = "Config";
 
-  private ImmutableList<Object> contexts;
-  private ImmutableList<Object> clusters;
-  private ImmutableList<Object> users;
+  private List<Object> contexts;
+  private List<Object> clusters;
+  private List<Object> users;
   private String currentContext;
 
   private KubeConfig() {}
 
   /** @return This config's contexts. */
-  public ImmutableList<Object> getContexts() {
+  public List<Object> getContexts() {
     return contexts;
   }
```

Comment:
I think the place the copy needs to be made is either in the builder (that calls `setContext` or `setContext`).  otherwise the list can change after it is created and the code will expect it to be unmodifiable - and it is not. (as a caller to `` can keep a reference to the list.)

Reply:
btw this break backward compat as well so I need to revert as well

Reply:
fixed